src/test/java/io/supertokens/test/accountlinking/GetUserByIdTest.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    AuthRecipeUserInfo createPasswordlessUserWithEmail(Main main, String email)
+            throws DuplicateLinkCodeHashException, StorageQueryException, NoSuchAlgorithmException, IOException,
+            RestartFlowException, InvalidKeyException, Base64EncodingException, DeviceIdHashMismatchException,
+            StorageTransactionLogicException, IncorrectUserInputCodeException, ExpiredUserInputCodeException {
+        Passwordless.CreateCodeResponse code = Passwordless.createCode(main, email, null,
+                null, "123456");
+        return Passwordless.consumeCode(main, code.deviceId, code.deviceIdHash,
+                code.userInputCode, null).user;
+    }
+
+    AuthRecipeUserInfo createPasswordlessUserWithPhone(Main main, String phone)
+            throws DuplicateLinkCodeHashException, StorageQueryException, NoSuchAlgorithmException, IOException,
+            RestartFlowException, InvalidKeyException, Base64EncodingException, DeviceIdHashMismatchException,
+            StorageTransactionLogicException, IncorrectUserInputCodeException, ExpiredUserInputCodeException {
+        Passwordless.CreateCodeResponse code = Passwordless.createCode(main, null, phone,
+                null, "123456");
+        return Passwordless.consumeCode(main, code.deviceId, code.deviceIdHash,
+                code.userInputCode, null).user;
+    }
+
+    @Test
+    public void testAllLoginMethods() throws Exception {
+        String[] args = {"../"};
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args, false);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{
+                        EE_FEATURES.ACCOUNT_LINKING, EE_FEATURES.MULTI_TENANCY});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        AuthRecipeUserInfo user1 = createEmailPasswordUser(process.getProcess(), "test@example.com", "password1");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user2 = createThirdPartyUser(process.getProcess(), "google", "userid1", "test@example.com");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user3 = createPasswordlessUserWithEmail(process.getProcess(), "test@example.com");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user4 = createPasswordlessUserWithPhone(process.getProcess(), "+919876543210");
+
+        assertFalse(user1.isPrimaryUser);
+        assertFalse(user2.isPrimaryUser);
+        assertFalse(user3.isPrimaryUser);
+        assertFalse(user4.isPrimaryUser);
+
+        assertEquals(user1, AuthRecipe.getUserById(process.getProcess(), user1.id));
+        assertEquals(user2, AuthRecipe.getUserById(process.getProcess(), user2.id));
+        assertEquals(user3, AuthRecipe.getUserById(process.getProcess(), user3.id));
+        assertEquals(user4, AuthRecipe.getUserById(process.getProcess(), user4.id));
```

Comment:
test structure of at least one of these.

Reply:
Done